java-manta-it/src/test/java/com/joyent/manta/client/helper/IntegrationTestHelper.java

```diff
@@ -73,10 +96,24 @@ public static void createTestBucketOrDirectory(final MantaClient client,
                                       final String testPath,
                                       final String testType) throws IOException {
         if ("buckets".equals(testType)) {
+            final String bucketPath = parseBucketPath(testPath);
             //Create bucket here
+            try {
+                client.createBucket(bucketPath);
+            } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.BUCKET_EXISTS_ERROR)) {
+                    LOG.error("Test Bucket named {} already exists leading to {}"
+                            , integrationTestBucket, e.getStatusMessage());
+                    e.setContextValue("Creating test bucket failed at :", testPath);
+                    throw e;
```

Comment:
Why are we throwing an exception if the bucket already exists? Don't we want the bucket to exist so that we can run the integration tests?

Reply:
We are throwing an exception since after each `integration-test` we clear up all the objects stored in the bucket and the bucket itself

Reply:
But it we want to delete the bucket only after the entire test-suite is completed that is also a viable option although my understanding regarding it was that it would mean if the integration-test suite is (interrupted/terminated) due to certain circumstances we will be left with residual test-data which won't be a feasible approach to invoke those in test environments like `staging` where these tests would be repeatedly run multiple times

Reply:
Maybe @cburroughs could answer my doubts here ?

Reply:
I would argue that we do not need to delete the bucket, but rather we should attempt to delete all of the files within the bucket that were created by the integration tests.

Reply:
Broadly, I would lean towards cleaning up at the end of the suite, not after every test, nor after failures (where one might need to go inspect the objects to understand what failed).  Cleaning up in general is good, but we will always need some sort of out of band 'garbage collection' to cover at least some cases.